diff --git a/node_modules/eslint/lib/rules/indent.js b/node_modules/eslint/lib/rules/indent.js
index 1556921..185d128 100644
--- a/node_modules/eslint/lib/rules/indent.js
+++ b/node_modules/eslint/lib/rules/indent.js
@@ -600,8 +600,10 @@ module.exports = {
 						properties: {
 							parameters: ELEMENT_LIST_SCHEMA,
 							body: {
-								type: "integer",
-								minimum: 0,
+								oneOf: [
+									{ type: "integer", minimum: 0 },
+									{ enum: ["keyword"] },
+								],
 							},
 						},
 						additionalProperties: false,
@@ -611,8 +613,10 @@ module.exports = {
 						properties: {
 							parameters: ELEMENT_LIST_SCHEMA,
 							body: {
-								type: "integer",
-								minimum: 0,
+								oneOf: [
+									{ type: "integer", minimum: 0 },
+									{ enum: ["keyword"] },
+								],
 							},
 						},
 						additionalProperties: false,
@@ -1325,6 +1329,23 @@ module.exports = {
 					blockIndentLevel = 1;
 				}
 
+				if ((node.parent && node.parent.type === "FunctionDeclaration" && options.FunctionDeclaration.body === "keyword") ||
+					(node.parent && (node.parent.type === "FunctionExpression" || node.parent.type === "ArrowFunctionExpression") && options.FunctionExpression.body === "keyword")) {
+					const fnKeyword = sourceCode.getFirstToken(node.parent);
+					const openBrace = sourceCode.getFirstToken(node);
+					const closeBrace = sourceCode.getLastToken(node);
+
+					offsets.setDesiredOffsets(
+						[openBrace.range[1], closeBrace.range[0]],
+						fnKeyword,
+						typeof blockIndentLevel === "number" ?
+							blockIndentLevel :
+							1,
+					);
+					offsets.setDesiredOffset(closeBrace, fnKeyword, 0);
+					return;
+				}
+
 				/*
 				 * For blocks that aren't lone statements, ensure that the opening curly brace
 				 * is aligned with the parent.
